tests/integration/ceph_auth_keystone_test.go

```diff
@@ -0,0 +1,192 @@
+)
+
+// ***************************************************
+// *** Major scenarios tested by the TestKeystoneAuthSuite ***
+// Setup
+// - A CephObject store will be created in a test cluster
+// Access
+// - Create a container
+// - Put a file in a container
+// - Get a file from the container
+// - Try to get the file without valid credentials
+// ***************************************************
+// *** Keystone/S3/Swift integration test implementation details ***
+// ceph_auth_keystone_test.go and ceph_base_keystone_test.go contain
+// integration tests for swift/s3 and keystone.
+// The tests are done against a minimal but functional setup of
+// OpenStack Keystone.
+// As it would be very complex to mock all of the involved keystone
+// functionality, the decision was made to use a working keystone setup
+// instead of implementing mocks.
+// The keystone container image used is provided by the Yaook-project
+// (see https://yaook.cloud).
+// The keystone configuration uses sqlite as the database backend.
+// cert-manager and trust-manager are used to provide the certificates
+// (cert-manager) and distribute the ca-certificate file(s) using a secret
+// (trust-manager).
+// To prevent hardcoded credentials, passwords are generated by the tests.
+// The integration tests use the aws s3 client (s3-functionality) and the
+// OpenStack CLI client (keystone- and swift-functionality).
+// (see https://docs.openstack.org/python-openstackclient/latest/).
+// This was a conscious design decision to use the standard client tooling
+// as close as possible to the used instead of using other go-libraries
+// (such as gophercloud) to (re-)implement the client tooling for the test.
+// ***************************************************
+
+func TestCephKeystoneAuthSuite(t *testing.T) {
+	s := new(KeystoneAuthSuite)
+	defer func(s *KeystoneAuthSuite) {
+		HandlePanics(recover(), s.TearDownSuite, s.T)
+	}(s)
+	suite.Run(t, s)
+}
+
+type KeystoneAuthSuite struct {
+	suite.Suite
+	helper    *clients.TestClient
+	installer *installer.CephInstaller
+	settings  *installer.TestCephSettings
+	k8shelper *utils.K8sHelper
+}
+
+func (h *KeystoneAuthSuite) SetupSuite() {
+	namespace := "keystoneauth-ns"
+	h.settings = &installer.TestCephSettings{
+		Namespace:         namespace,
+		OperatorNamespace: namespace,
+		StorageClassName:  "",
+		UseHelm:           true,
+		UsePVC:            false,
+		Mons:              1,
+		SkipOSDCreation:   false,
+		// EnableAdmissionController: true,
+		EnableDiscovery:      true,
+		ChangeHostName:       true,
+		ConnectionsEncrypted: true,
+		RookVersion:          installer.LocalBuildTag,
+		CephVersion:          installer.QuincyVersion,
+		SkipClusterCleanup:   false,
+		SkipCleanupPolicy:    false,
+	}
+	h.settings.ApplyEnvVars()
+	h.installer, h.k8shelper = StartTestCluster(h.T, h.settings)
+
+	// install yaook-keystone here
+	if err := InstallKeystoneInTestCluster(h.k8shelper, namespace); err != nil {
+		return
+	}
```

Comment:
In this case, shouldn't the test fail, instead of returning as though this ran successfully?

Reply:
Addressed in 64eafbac648ef8d80784561b51f950bc2f85d9fa